### ISSUE TITLE
UnsupportedOperationException with unlocatable hosts config file

### ISSUE THREAD

**Issue by Kevin-A** [eclipse/vert.x#1517]
Hi,

Thank you for taking your time to read my entry. Last night while work was being done on the environments, we got this error:

```
java.lang.UnsupportedOperationException: null
        at java.util.AbstractMap.put(AbstractMap.java:209) ~[na:1.8.0_45]
        at io.vertx.core.impl.AddressResolver.<init>(AddressResolver.java:115) ~[jar-with-dependencies.jar:na]
        at io.vertx.core.impl.VertxImpl.<init>(VertxImpl.java:152) ~[jar-with-dependencies.jar:na]
        at io.vertx.core.impl.VertxImpl.<init>(VertxImpl.java:119) ~[jar-with-dependencies.jar:na]
        at io.vertx.core.impl.VertxImpl.<init>(VertxImpl.java:115) ~[jar-with-dependencies.jar:na]
        at io.vertx.core.impl.VertxFactoryImpl.vertx(VertxFactoryImpl.java:34) ~[jar-with-dependencies.jar:na]
        at io.vertx.core.Vertx.vertx(Vertx.java:80) ~[jar-with-dependencies.jar:na]
        at com.something.something.Main.main(Main.java:108) ~[jar-with-dependencies.jar:na]
```

We suspect that it is due to the following:

In the `AddressResolver` class on line 109 (https://github.com/eclipse/vert.x/blob/HEAD/src/main/java/io/vertx/core/impl/AddressResolver.java#L109) the parseSilently returns a `Collections.emptyMap()` as it probably has trouble finding the hosts file. Then, on line 115 (https://github.com/eclipse/vert.x/blob/master/src/main/java/io/vertx/core/impl/AddressResolver.java#L115) an attempt is made to `put` a value into this map, which is not a supported action.

So, the issue is caused by attempting to put a value in a map that does not support puts

Extra:
- Vertx 3.2.1
- rhel7

I might have missed some information, if so, please request and I'll do my best to provide it.

**Comment by vietj** [comment]
shall be fixed in 3.3.3

**Comment by vietj** [comment]
not easy to test though
